bip-0077.md

```diff
@@ -0,0 +1,703 @@
+receiver-specified expiration time.
+
+### 64-bit Short ID Length
+
+64 bits are sufficient to make the probability of experiencing a random
+collision negligible. As of writing, the UTXO set has ~2^28 elements.
+This is a very loose upper bound for the number of concurrent (non-spam)
+sessions, for which the probability of a random collision will be less
+than 1%. The actual number of sessions will of course be (orders of
+magnitudes) lower given that sessions are short lived. With ~2^21
+sessions (a loose bound on number of transactions that can be confirmed
+in 24 hours) the probability is less than 1e-6. These figures bound the
+probability of a collision existing anywhere in the entire set, whereas
+the probability for an individual session to experience a collision is
+\<\< 1e-10 in either case.
+
+### Complete UTXO Data
+
+Complete UTXO data is required because this information is required for
+signing and calculating fees for some input types.
+
+### HTTP
+
+HTTP is ubiquitous. Using simple HTTP polling allows even Bitcoin Core
+to consider an implementation. Unlike a WebSockets protocol, plain HTTP
+can benefit from metadata protection by using Oblivious HTTP.
+
+### Oblivious HTTP
+
+OHTTP protects sender and receiver IP addresses both from one another
+and from the directory. This makes it more difficult for a directory to
+correlate many payjoin transactions with specific IP addresses.
+
+OHTTP relays can be run as basic HTTP proxies from wallet providers or
+third parties.
+
+### Uniform Payloads
+
+Encapsulated OHTTP payloads seen by the relay and directory, and
+encrypted messages seen by the directory, are constructed to be uniform
+so that these third-party services are unable to distinguish between
+them.
+
+Encapsulated OHTTP messages are 8192 bytes long, and begin with a
+cleartext OHTTP header and an uncompressed key which is distinguishable
+from random bytes but uniform across different encapsulated requests.
+
+End-to-end encrypted messages are 7168 bytes long, and should be
+indistinguishable from uniformly random bytes.
+[ElligatorSwift as defined in BIP 324](https://github.com/bitcoin/bips/blob/master/bip-0324.mediawiki#elligatorswift-encoding-of-curve-x-coordinates)
+is used to encode encapsulated HPKE public keys prepended to the HPKE ciphertext
+so that the directory can't distinguish between key material, the
+ciphertext, and randomness. This ensures the two different protocol
+messages are indistinguishable from each other as well as any protocol
+extensions.
+
+These padded sizes are sufficient for most PSBTs without exceeding the [
+8KB
+limit](https://www.geekersdigest.com/max-http-request-header-size-server-comparison/)
+of many HTTP/1.1 web servers. 8KB is also too small for image sharing,
+making misuse of the directory impractical.
+
+### Secp256k1 Hybrid Public Key Encryption
+
+[RFC 9180 Hybrid Public Key
+Encryption](https://www.rfc-editor.org/rfc/rfc9180.html)
```

Comment:
This section and perhaps the next one (Secp256k1 based DHKEM) may need to be more detailed.

[RFC 9180 (Section 7.1)](https://www.ietf.org/rfc/rfc9180.html#section-7.1) lists the initial set of registered KEM identifiers. These include DHKEMs based on:
* NIST curves: P-256, P-384, P-521
* Curve25519, Curve448
* Crucially, secp256k1 is not among the curves for which KEMs are pre-defined in RFC 9180.

Instead of the 'Secp256k1-based DHKEM' section of this BIP pointing to a draft link, it could describe the scheme directly here

Reply:
I'd rather post to the draft, which [IANA also links to as the authoritative reference](https://www.iana.org/assignments/hpke/hpke.xhtml) rather than try to copy it.

I also find the sentence structure "x may need to be more detailed" confusing because I don't know what inspired the comment. Do you have a reason why you prefer the scheme??